### PR TITLE
Add speedtest

### PR DESCRIPTION
**Pull request by tonka3000**
## Description

Add speed test via speedtest.net cli. The speedtest.net cli is automatically downloaded and will get cache (cli is only about 600KB).

License agreements will be automatically accepted. Details about the license agreements of speedtest.net cli is added to the README.md file.

## Type of change

- New extension

### Running speedtest
https://user-images.githubusercontent.com/3163807/140605799-518da6df-8678-449d-9c39-c0e0cc3cb9d4.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

File: extensions/speedtest/src/lib/utils.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    while (bits >= 2000 && unit < 4) {
+        unit++;
+        bits /= 1000;
+    }
+    return `${bits.toFixed(places[unit])} ${units[unit]}bps`;
+}
+
+export async function fileExists(file: string): Promise<boolean> {
+    return new Promise(resolve => fs.access(file, fs.constants.F_OK, err => resolve(!err)));
+}
```
Comment (pernielsentikaer) [inline extensions/speedtest/src/lib/utils.ts:25]:
```suggestion
```

Unused function, should be removed

File: extensions/speedtest/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          },
+          (r: Result) => {
+            if (!cancel) {
+              setResult({ ...r });
+              setIsLoading(false);
+            }
+          },
+          (err: Error) => {
+            if (!cancel) {
+              setError(err instanceof Error ? err.message : "unknown error");
```
Comment (pernielsentikaer) [inline extensions/speedtest/src/index.tsx:138]: Redundant instanceof check

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @tonka3000 👋 

Awesome extension 💯 

Only a few suggestions left before we're ready to merge 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @tonka3000 👋 

You're on fire 🔥 

- A thing I forgot to mention last time was that the "Copy to Clipboard" should maybe be a summary of the speedtest instead of just the list.item value.

- Just a thought, is it possible to add how many percentages completed, somehow so the user has an idea about how long time it takes, if the internet speed is pretty stable it feels like it's stuck.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by pernielsentikaer** [comment]
Hi @tonka3000 👋 

It looks good, but I will suggest using the subtitle in the list items for it instead, it might be more suitable. Do you think that's possible to implement? After that, I think we're ready to merge it 💪 😄

**Comment by tonka3000** [comment]
@pernielsentikaer 

> It looks good, but I will suggest using the subtitle in the list items for it instead, it might be more suitable. Do you think that's possible to implement? After that, I think we're ready to merge it 💪 😄

Do you refer to the progress bar?

**Comment by PitNikola** [comment]
@tonka3000, how does it work on M1? It runs under Rosetta?

**Comment by pernielsentikaer** [comment]
> Do you refer to the progress bar (%)?

Yes, the searchBarPlaceholder - could we move it to the corresponding item instead when it's running?

@PitNikola I'm on M1 and having no problems at all 😄

**Comment by tonka3000** [comment]
@PitNikola 

> @tonka3000, how does it work on M1? It runs under Rosetta?

Is also don't have a M1, but somebody on slack already tested it on M1. So it run at least on 2 M1 machines 😄 .

**Comment by PitNikola** [comment]
Ok, then it's probably Rosetta. Ok, let's address that comment from Per about moving % progress info from the placeholder to subtitles of list items, and it's ready for merging.

Really nice extension, you've been quite creative! 🧑‍🎨

**Comment by tonka3000** [comment]
@PitNikola @pernielsentikaer 

https://user-images.githubusercontent.com/3163807/141351945-0973f9f4-9fc3-42a8-9020-2d4ac8cbd42d.mp4

**Comment by PitNikola** [comment]
@tonka3000, how about hiding "100%" after it completes. Seems redundant, WDYT?

**Comment by tonka3000** [comment]
@PitNikola 

https://user-images.githubusercontent.com/3163807/141358495-0b8e444b-eb5a-4d0a-8aea-78844225b18e.mp4

**Comment by pernielsentikaer** [comment]
Much better 💯 Tested everything again with great success 💪 

Let's get it merged, as @PitNikola said it's very creative 🥇

**Comment by PitNikola** [comment]
Nice! If ping is often going to be so fast, I would consider not showing the progress for it. Feel free to address it in future PRs.

**Comment by PitNikola** [comment]
Also another suggestion: show HUD toast after copying summary saying something like "Summary copied to clipboard"

**Comment by pernielsentikaer** [comment]
I'm pretty sure that it's already showing a message

EzA6HitllPljfWBe

**Comment by PitNikola** [comment]
Ah, right. Hm, maybe I had some background where it was less visible...

**Pull request merged**